File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> umap;
        int n = nums.size();
        int res = 0;

        for( auto p : nums){
            umap[p]++;
        }

        for( auto &p : umap){
            if( p.second > n/2){
                res = p.first;
                return res;
            }
        }
return -1;
        
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate = None
        votes = 0

        for n in nums: 
            if(votes == 0):
                candidate = n

            if(candidate == n):
                votes += 1

            else: 
                votes -= 1

        return candidate